src/app/api/advocates/route.ts

```diff
@@ -1,12 +1,45 @@
+import { NextRequest, NextResponse } from "next/server";
 import db from "../../../db";
 import { advocates } from "../../../db/schema";
-import { advocateData } from "../../../db/seed/advocates";
+import { and, gte, like } from "drizzle-orm";
 
-export async function GET() {
-  // Uncomment this line to use a database
-  // const data = await db.select().from(advocates);
+export async function GET(req: NextRequest) {
+  console.log("inside GET");
 
-  const data = advocateData;
+  try {
+    const { searchParams } = new URL(req.url);
 
-  return Response.json({ data });
+    const lastName = searchParams.get("lastName");
+    const firstName = searchParams.get("firstName");
+    const city = searchParams.get("city");
+    const degree = searchParams.get("degree");
+    const specialty = searchParams.get("specialty");
+    const yearsOfExperience = searchParams.get("yearsOfExperience");
+
+    const whereClause = [];
+
+    if (firstName) whereClause.push(like(advocates.firstName, firstName + "%"));
+    if (lastName) whereClause.push(like(advocates.lastName, lastName + "%"));
+    if (city) whereClause.push(like(advocates.city, city + "%"));
+    if (degree) whereClause.push(like(advocates.degree, degree));
+    if (yearsOfExperience)
+      whereClause.push(gte(advocates.yearsOfExperience, yearsOfExperience));
+
+    const data = await db
+      .select()
+      .from(advocates)
+      .where(and(...whereClause));
+
+    const filteredData = await specialty
```

Comment:
We're not actually searching the database for specialty since there is a lot of data jammed into this field.  If we need to search this field, then we should pull these out into some kind of a search index.